apps/shell/src/app/modules/settings/pages/settings-general/github-conenction/github-connection.tsx

```diff
@@ -38,9 +41,13 @@ export function GithubConnection() {
   useEffect(() => {
     const code = searchParams.get('code');
     const state = searchParams.get('state');
-    if (!code || !state) return;
+    if (!code || !state) {
+      document.cookie = `gh_state=${formState}; path=/`;
+      return;
+    }
     connectGitHub();
-  }, [connectGitHub, searchParams]);
+    window.history.pushState({}, '', `/settings`);
```

Comment:
You'll need to use React Router DOM's navigation hook instead. Otherwise this can circumvent a number of hooks and guards in the application.

Include the hook.

```
export function GithubConnection() {
  const navigate = useNavigate();
```

Then call it like this here.

```suggestion
    navigate(`/settings`);
```